src/lib/dai.c

```diff
@@ -136,26 +137,56 @@ static inline const struct dai_type_info *dai_find_type(uint32_t type)
 	return NULL;
 }
 
-struct dai *dai_get(uint32_t type, uint32_t index, uint32_t flags)
+#ifdef __ZEPHYR__
+static struct dai *dai_get_zephyr(const struct dai_type_info *dti, uint32_t type, uint32_t index)
 {
-	int ret = 0;
-	const struct dai_type_info *dti;
 	struct dai *d;
-	k_spinlock_key_t key;
+	int ret;
 
-	dti = dai_find_type(type);
-	if (!dti)
-		return NULL; /* type not found */
+	tr_info(&dai_tr,"log_ssp_data get_dai_zephyr index: %u", index);
+
+	for (d = dti->dai_array; d < dti->dai_array + dti->num_dais; d++) {
+		if (d->index != index)
+			continue;
+
+		ret = dai_probe_op(d);
+
+		return !ret ? d : NULL;
```

Comment:
positive tests usually require a couple of brain cycles fewer to process, i.e. `return ret ? NULL : d;` might be a bit faster to read. In fact even
```
if (d->index == index)
	return dai_probe_op(d) ? NULL : d;
```

Reply:
oh yes, will change to that